src/values.rs

```diff
@@ -257,8 +311,52 @@ impl PartialOrd for ExtNumeric {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct Binary {
+   /// The actual data. We put it behind an `Arc` to make sure
+   /// that cloning remains inexpensive.
+   data: Arc<Vec<u8>>,
+   /// The mime type. Should probably be an Id<MimeTypeId>.
+   mimetype: String
+}
+
 /// Representation of an actual value that can be sent to/received
 /// from a service.
+///
+/// # (De)serialization
+///
+/// Values of this state are represented by an object `{ key: value }`, where key is one of
+/// `Unit`, `OnOff`, `OpenClosed`, ... Note that the `value` for `Unit` is `[]`.
+///
+/// ```
+/// extern crate serde;
+/// extern crate serde_json;
+/// extern crate foxbox_taxonomy;
+///
+/// # fn main() {
```

Comment:
__question:__ mostly for myself, what the purpose of ```# fn main() {``` in rust doc? Does it affect somehow generated documentation?